clients/src/main/java/org/apache/kafka/clients/consumer/ShareConsumer.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer;
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @see KafkaShareConsumer
+ * @see MockShareConsumer
+ */
+@InterfaceStability.Evolving
+public interface ShareConsumer<K, V> extends Closeable {
+
+    /**
+     * @see KafkaShareConsumer#subscription()
+     */
+    Set<String> subscription();
+
+    /**
+     * @see KafkaShareConsumer#subscribe(Collection)
+     */
+    void subscribe(Collection<String> topics);
+
+    /**
+     * @see KafkaShareConsumer#unsubscribe()
+     */
+    void unsubscribe();
+
+    /**
+     * @see KafkaShareConsumer#poll(Duration)
+     */
+    ConsumerRecords<K, V> poll(Duration timeout);
+
+    /**
+     * @see KafkaShareConsumer#acknowledge(ConsumerRecord)
+     */
+    void acknowledge(ConsumerRecord<K, V> record);
+
+    /**
+     * @see KafkaShareConsumer#acknowledge(ConsumerRecord, AcknowledgeType)
+     */
+    void acknowledge(ConsumerRecord<K, V> record, AcknowledgeType type);
+
+    /**
+     * @see KafkaShareConsumer#commitSync()
+     */
+    Map<TopicIdPartition, Optional<KafkaException>> commitSync();
```

Comment:
Though it's debatable and not sure if wrapping the exception here (as method params) would be beneficial as with Optional as well we need a null check regarding if optional is not null.

Reply:
It will contain `Optional.empty()` if there's no exception.

Reply:
Sure, just thinking if Optional<KafkaException> vs KafkaException as param, though I know client code has Optional params and I added similar for KIP-714 but sometimes I think it's unnecessary wrapping.